lib/android-helpers.js

```diff
@@ -628,7 +628,12 @@ helpers.verifyUnlock = async function (adb) {
 };
 
 helpers.initDevice = async function (adb, opts) {
+  if (opts.skipDeviceInitialization) {
+    logger.info(`'skipDeviceInitialization' is set. Skipping device initialization.`);
+    return await adb.defaultIME();
+  }
   await adb.waitForDevice();
```

Comment:
I'm also not quite sure about the unicode keyboard. Isn't it reset on session deletion?

Reply:
It is, but what we skip is change of the keyboard:
```
  logger.debug(`Unsetting previous IME ${defaultIME}`);
  const appiumIME = `${SETTINGS_HELPER_PKG_ID}/.UnicodeIME`;
  logger.debug(`Setting IME to '${appiumIME}'`);
  await adb.enableIME(appiumIME);
  await adb.setIME(appiumIME);
```
the return value stays the same, so if we have valid keyboard at the beginning, then the behavior should be the same.

Reply:
> And if we want to skip initialization, then we assume that device is ready for them and we don't want to change it's state.

It might not be what a user wants. Providing this option will render the `unicodeKeyboard` capability invalid, because the IME is not going to be changed to the unicode one.

Reply:
That's good point. I'll change that.

Reply:
Done.